qutebrowser/misc/mhtml.py

```diff
@@ -0,0 +1,411 @@
+
+
+_CSS_URL_PATTERNS = [re.compile(x) for x in [
+    rb"@import '(?P<url>[^']+)'",
+    rb'@import "(?P<url>[^"]+)"',
+    rb'''url\((?P<url>[^'"][^)]*)\)''',
+    rb'url\("(?P<url>[^"]+)"\)',
+    rb"url\('(?P<url>[^']+)'\)",
+]]
+
+
+def _get_css_imports(data):
+    """Return all assets that are referenced in the given CSS document.
+
+    The returned URLs are relative to the stylesheet's URL.
+
+    Args:
+        data: The content of the stylesheet to scan.
+    """
+    if isinstance(data, str):
+        data = data.encode("utf-8")
+    urls = []
+    for pattern in _CSS_URL_PATTERNS:
+        for match in pattern.finditer(data):
+            url = match.group("url")
+            if url:
+                urls.append(url)
+    return urls
+
+
+MHTMLPolicy = policy.default.clone(linesep="\r\n", max_line_length=0)
+
+
+def _chunked_base64(data, maxlen=76, linesep=b"\r\n"):
+    """Just like b64encode, except that it breaks long lines.
+
+    Args:
+        maxlen: Maximum length of a line, not including the line separator.
+        linesep: Line separator to use as bytes.
+    """
+    encoded = b64encode(data)
+    result = []
+    for i in range(0, len(encoded), maxlen):
+        result.append(encoded[i:i + maxlen])
+    return linesep.join(result)
+
+
+def _rn_quopri(data):
+    """Return a quoted-printable representation of data."""
+    # See RFC 2045 https://tools.ietf.org/html/rfc2045#section-6.7
+    # The stdlib version in the quopri module has inconsistencies with line
+    # endings and breaks up character escapes over multiple lines, which isn't
+    # understood by qute and leads to jumbled text
+    maxlen = 76
+    whitespace = {ord(b"\t"), ord(b" ")}
+    output = []
+    current_line = b""
+    for byte in data:
+        # Literal representation according to (2) and (3)
+        if (ord(b"!") <= byte <= ord(b"<") or ord(b">") <= byte <= ord(b"~")
+                or byte in whitespace):
+            current_line += bytes([byte])
+        else:
+            current_line += b"=" + "{:02X}".format(byte).encode("ascii")
+        if len(current_line) >= maxlen:
+            # We need to account for the = character
+            split = [current_line[:maxlen - 1], current_line[maxlen - 1:]]
+            quoted_pos = split[0].rfind(b"=")
+            if quoted_pos + 2 >= maxlen - 1:
+                split[0], token = split[0][:quoted_pos], split[0][quoted_pos:]
+                split[1] = token + split[1]
+            current_line = split[1]
+            output.append(split[0] + b"=")
+    output.append(current_line)
+    return b"\r\n".join(output)
+
+
+E_NONE = (None, lambda x: x)
+"""No transfer encoding, copy the bytes from input to output"""
+
+E_BASE64 = ("base64", _chunked_base64)
+"""Encode the file using base64 encoding"""
+
+E_QUOPRI = ("quoted-printable", _rn_quopri)
+"""Encode the file using MIME quoted-printable encoding."""
+
+
+class MHTMLWriter(object):
```

Comment:
No need to inherit from `object` as this is Python 3 - for consistency, this should be removed (same for the other classes in this file).